docs/architecture/index.rst

```diff
@@ -1,25 +1,586 @@
+| Checkpointing | *Checkpointing* captures the training state at a given time in the   |
+|               | *model registry*.                                                    |
++---------------+----------------------------------------------------------------------+
+| Scheduling    | *Scheduling* schedules jobs to run, ensuring that all of the compute |
+|               | resources required for a job are available before the job launches.  |
++---------------+----------------------------------------------------------------------+
+
+Agent Node Scope of Responsibility
+----------------------------------
+
+Agent node responsibilities:
+
+-  Discovers local computing devices and sends device/slot metadata to the master.
+-  Runs workloads at the request of the master.
+-  Monitors containers and sends container information to the master.
+-  For a *trial runner*, which runs a trial in a containerized environment, reports trial runner
+   states to the master.
+
+The agent manages a number of GPU or CPU devices, which are referred to as *slots*. There is
+typically one agent per compute server and the active experiment volume dictates the number of
+agents needed.
+
+Agents communicate only with the master and state information is not maintained on the agent.
+
+PostgreSQL Database
+===================
+
+Each cluster requires access to a `PostgreSQL <https://www.postgresql.org/>`_ database to store
+experiment and trial metadata. Although not required, the database typically resides on the master.
+If you use the ``det deploy`` command to launch Determined on a cloud provider or on-premises,
+PostgreSQL preinstalls a network file system shared by all agents. Otherwise, you need to manually
+install PostgreSQL.
+
+Docker Images
+=============
+
+Determined launches workloads using `Docker <https://www.docker.com/>`_ containers. By default,
+workloads execute inside a Determined-provided container that includes common deep learning
+libraries and frameworks.
+
+Default Docker images are provided to launch containers for experiments, commands, and other
+workflows. All trial runner containers are launched with additional Determined-specific harness
+code, which orchestrates model training and evaluation in the container. Trial runner containers are
+also loaded with the experiment model definition and hyperparameter values for the current trial.
+GPU-specific versions of each library are automatically selected when running on agents with GPUs.
+
+If your model code has additional dependencies, you can specify a startup hook to load the
+additional dependencies. If a startup hook exists, the file automatically runs with every Docker
+container startup.
+
+If you use the ``det deploy`` command on a cloud provider, Docker is preinstalled. For a manual or
+on-premises deployment using ``det deploy``, you need to manually install Docker.
+
+Additional Core Cloud Resources
+===============================
```

Comment:
This section should be omitted or further explained.

Right now, it looks like we are just listing some well-known features of a couple of cloud platforms.